File: llm/lib/tasks/draft-concept.mjs
// draft-concept — draft a concept page from sources.
import { runJsonPrompt } from '../runner.mjs';

export async function run({ kbRoot, input }) {
  const slug = input?.slug || 'stub-concept';
  const sources = Array.isArray(input?.sources) ? input.sources : [];
  const related = Array.isArray(input?.related) ? input.related : [];
  const { data } = await runJsonPrompt(kbRoot, 'draft-concept', {
    slug,
    sources: sources.join('\n\n---\n\n'),
    related: related.map((r) => `- ${r}`).join('\n'),
  });
  return {
    task: 'draft-concept',
    slug: data.slug || slug,
    title: data.title || slug,
    body: data.body || '',
    sources: Array.isArray(data.sources) ? data.sources : sources,
  };
}

File: agent/weft_agent/tasks/draft_concept.py
"""draft-concept — draft a concept page from sources (port of llm/lib/tasks/draft-concept.mjs)."""
from ..runner import run_json_prompt


def run(kb_root, input=None, output_path=None):
    input = input or {}
    slug = input.get("slug") or "stub-concept"
    sources = input.get("sources") if isinstance(input.get("sources"), list) else []
    related = input.get("related") if isinstance(input.get("related"), list) else []
    res = run_json_prompt(kb_root, "draft-concept", {
        "slug": slug,
        "sources": "\n\n---\n\n".join(str(s) for s in sources),
        "related": "\n".join(f"- {r}" for r in related),
    })
    data = res["data"]
    return {
        "task": "draft-concept",
        "slug": data.get("slug") or slug,
        "title": data.get("title") or slug,
        "body": data.get("body") or "",
        "sources": data.get("sources") if isinstance(data.get("sources"), list) else sources,
    }
